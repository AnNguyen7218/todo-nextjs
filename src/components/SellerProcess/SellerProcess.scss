@import 'config.gutters';
@import 'config.media';

.container {
  margin-bottom: gutter(32);

  @include break(md) {
    margin-bottom: gutter(0);
  }
}

.img {
  width: 21rem;
  height: 21rem;
  margin-bottom: gutter(20);

  @include break(md) {
    width: 12rem;
    height: 12rem;
    margin-bottom: gutter(16);
  }
}

.title {
  margin-bottom: gutter(4);

  @include break(md) {
    margin-bottom: gutter(8);
  }

  @include break(xl) {
    margin-bottom: gutter(4);
  }
}

.description {
  width: 100%;

  @include break(md) {
    min-width: 20.8rem;
  }

  @include break(xl) {
    width: 100%;
  }
}
