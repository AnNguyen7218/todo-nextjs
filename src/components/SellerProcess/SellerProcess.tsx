import React from 'react';
import css from './SellerProcess.scss';

type Props = {
  process: SellerProcessType;
};

const SellerProcess: React.FC<Props> = ({
  process: { src, title, description },
}) => {
  return (
    <div className={css['container']}>
      <img src={src} alt="process" className={css['img']} />
      <h3 className={css['title']}>{title}</h3>
      <p className={css['description']}>{description}</p>
    </div>
  );
};

export default SellerProcess;
