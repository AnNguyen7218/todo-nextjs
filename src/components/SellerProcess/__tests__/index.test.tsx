import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Step1 from 'assets/images/step1.png';
import SellerProcess from '../SellerProcess';

afterEach(cleanup);

const process = {
  src: Step1,
  title: '01. Nec porta erat tinc',
  description:
    'In in ultricies eget quam suspendisse metus. Fringilla in vulputate malesuada interdum. Pulvinar eget tempor',
};

describe('SellerProcess Component', () => {
  test('SellerProcess should match snapshot', () => {
    const { container } = render(<SellerProcess process={process} />);
    expect(container).toMatchSnapshot();
  });
});
