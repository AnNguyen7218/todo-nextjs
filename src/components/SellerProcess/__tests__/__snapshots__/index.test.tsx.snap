// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SellerProcess Component SellerProcess should match snapshot 1`] = `
<div>
  <div
    class="container"
  >
    <img
      alt="process"
      class="img"
      src="step1.png"
    />
    <h3
      class="title"
    >
      01. Nec porta erat tinc
    </h3>
    <p
      class="description"
    >
      In in ultricies eget quam suspendisse metus. Fringilla in vulputate malesuada interdum. Pulvinar eget tempor
    </p>
  </div>
</div>
`;
